cranelift/codegen/src/isa/zkasm/inst/mod.rs

```diff
@@ -433,21 +433,10 @@ fn zkasm_get_operands<F: Fn(VReg) -> VReg>(inst: &Inst, collector: &mut OperandC
                     .expect("Only support 1 register in comparison now"),
                 b0(),
             );
-            if cc == IntCC::NotEqual {
-                // zkasm don't have NEQ opcode so we need a bit more computation
-                let mut clobbered = PRegSet::empty();
-
-                if rd.to_reg() == a0() {
-                    clobbered.add(b0().to_real_reg().unwrap().into());
-                } else if rd.to_reg() == b0() {
-                    clobbered.add(a0().to_real_reg().unwrap().into());
-                } else {
-                    clobbered.add(a0().to_real_reg().unwrap().into());
-                    clobbered.add(b0().to_real_reg().unwrap().into());
-                }
-                collector.reg_clobbers(clobbered);
-            }
-            collector.reg_def(rd);
+            let mut clobbers = PRegSet::empty();
+            clobbers.add(b0().to_real_reg().unwrap().into());
+            collector.reg_clobbers(clobbers);
+            collector.reg_fixed_def(rd, a0());
```

Comment:
The clobber here is only required for `NotEqual` variant still. For other conditional branches, the `B` retains its original value after the comparison, does it not?

I’m somewhat concerned about keeping the two pieces of code in sync, but I can’t think of a good way to avoid the duplication so it is fine for now.

Reply:
Fixed [here](https://github.com/near/wasmtime/pull/57/commits/8ffaf9136f0ebd1db5ebc0f43bade56d99bb537f), sorry